packages/core/admin/admin/src/content-manager/pages/EditView/components/DocumentActions.tsx

```diff
@@ -431,13 +447,28 @@ const PublishAction: DocumentActionComponent = ({ activeTab, id, model, collecti
   const isSubmitting = useForm('PublishAction', ({ isSubmitting }) => isSubmitting);
   const document = useForm('UpdateAction', ({ values }) => values);
 
+  const isDocumentPublished =
+    document?.[PUBLISHED_AT_ATTRIBUTE_NAME] ||
+    meta?.availableStatus.some((doc) => doc[PUBLISHED_AT_ATTRIBUTE_NAME] !== null);
+
   return {
+    /**
+     * Disabled when:
+     *  - the form is submitting
+     *  - the active tab is the published tab
+     *  - the document is already published & not modified
+     *  - the document is being created & not modified
+     *  - the user doesn't have the permission to publish
+     *  - the user doesn't have the permission to create a new document
+     *  - the user doesn't have the permission to update the document
+     */
     disabled:
-      !canPublish ||
       isSubmitting ||
       activeTab === 'published' ||
-      !modified ||
-      Boolean((!id && !canCreate) || (id && !canUpdate)),
+      (!modified && isDocumentPublished) ||
+      (!modified && !document.id) ||
+      !canPublish ||
+      Boolean((!document.id && !canCreate) || (document.id && !canUpdate)),
```

Comment:
That is one hell of a condition 🤯

Reply:
yeah i've put a comment above it now to explain how it works :)